src/commonMain/kotlin/org/kosat/CDCL.kt

```diff
@@ -3,20 +3,19 @@ package org.kosat
 import kotlin.math.abs
 import kotlin.math.max
 
-// CDCL
-fun solveCnf(cnf: CnfRequest): List<Int>? {
-    val clauses = (cnf.clauses.map { it.lit }).toMutableList()
-    val solver = Kosat(clauses, cnf.vars)
-    return if (solver.solve()) solver.getModel() else null
-}
+abstract class CDCL(val clauses: MutableList<MutableList<Int>>, initNumber: Int = 0) {
```

Comment:
Maybe embrace the "composition over inheritance" well-known principle? Abstract classes will bite you hard, I would suggest avoiding them if possible. And here it is very much possible: `CDCL` is just "a box with functions implementing the CDCL algorithm", you don't *have to* inherit it: just make `val internal: CDCL` in your external code and delegate calls to it when necessary.